x/ecocredit/server/core/create_project.go

```diff
@@ -80,3 +86,32 @@ func (k Keeper) genProjectID(ctx context.Context, classKey uint64, classID strin
 
 	return core.FormatProjectId(classID, nextSeq), nil
 }
+
+// verifyReferenceId prevents multiple projects from having the same reference id within the
+// scope of a credit class. We verify this here at the message server level rather than at the
+// ORM level because reference id is optional and therefore multiple projects within the scope
+// of a credit class can have an empty reference id (see BridgeReceive for more information)
+func (k Keeper) verifyReferenceId(ctx context.Context, classKey uint64, referenceId string) error {
+	if referenceId == "" {
+		// reference id is optional so an empty reference id is valid
+		return nil
+	}
+
+	key := api.ProjectClassKeyReferenceIdIndexKey{}.WithClassKeyReferenceId(classKey, referenceId)
+	it, err := k.stateStore.ProjectTable().List(ctx, key)
+	if err != nil {
+		return err
+	}
+	var total int
+	if it.Next() {
+		total++
+	}
+	it.Close()
+	if total > 0 {
+		return sdkerrors.ErrInvalidRequest.Wrapf(
+			"a project with reference id %s already exists within this credit class", referenceId,
+		)
+	}
```

Comment:
```suggestion
        defer it.Close()
	if it.Next() {
		return sdkerrors.ErrInvalidRequest.Wrapf(
			"a project with reference id %s already exists within this credit class", referenceId,
		)
	}
```

can simplify a bit here, `it.Next()` will return `true` if theres at least 1 item in the iterator